src/arm/windows/init_by_logical_sys_info.c

```diff
@@ -0,0 +1,877 @@
+							global_proc_index_per_group,
+							info,
+							info_id);
+			break;
+			default:
+				cpuinfo_log_error(
+					"unexpected processor info type (%"PRIu32") for processor information",
+					(uint32_t) info->Relationship);
+				result = false;
+			break;
+		}
+		if (!result) {
+			nr_of_structs = 0;
+			goto clean_up;
+		}
+	}
+clean_up:
+	/* 5. Release dynamically allocated info structure. */
+	HeapFree(heap, 0, infos);
+	infos = NULL;
+	return nr_of_structs;
+}
+
+static bool parse_relation_processor_info(
+	struct cpuinfo_processor* processors,
+	uint32_t nr_of_processors,
+	const uint32_t* global_proc_index_per_group,
+	PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX info,
+	const uint32_t info_id,
+	struct cpuinfo_core* cores,
+	const struct woa_chip_info *chip_info)
+{
+	for (uint32_t i = 0; i < info->Processor.GroupCount; i++) {
+		const uint32_t group_id = info->Processor.GroupMask[i].Group;
+		/* Bitmask representing processors in this group belonging to this package */
+		KAFFINITY group_processors_mask = info->Processor.GroupMask[i].Mask;
+		while (group_processors_mask != 0) {
+			const uint32_t processor_id_in_group =
+				low_index_from_kaffinity(group_processors_mask);
+			const uint32_t processor_global_index =
+				global_proc_index_per_group[group_id] + processor_id_in_group;
+
+			if(processor_global_index >= nr_of_processors) {
+				cpuinfo_log_error("unexpected processor index %"PRIu32"",
+					processor_global_index);
+				return false;
+			}
+
+			switch(info->Relationship) {
+				case RelationProcessorPackage:
+					store_package_info_per_processor(
+						processors, processor_global_index, info_id,
+						group_id, processor_id_in_group,
+						chip_info);
+				break;
+				case RelationProcessorCore:
+					store_core_info_per_processor(
+						processors, processor_global_index,
+						info_id, info,
+						cores, chip_info);
+				break;
+				default:
+					cpuinfo_log_error(
+						"unexpected processor info type (%"PRIu32") for processor information",
+						(uint32_t) info->Relationship);
+				break;
+			}
+			/* Clear the bits in affinity mask, lower the least set bit. */
+			group_processors_mask &= (group_processors_mask - 1);
+		}
+	}
+	return true;
+}
+
+static bool parse_relation_cache_info(
+	struct cpuinfo_processor* processors,
+	uint32_t nr_of_processors,
+	struct cpuinfo_cache* caches,
+	uint32_t* numbers_of_caches,
+	const uint32_t* global_proc_index_per_group,
+	PSYSTEM_LOGICAL_PROCESSOR_INFORMATION_EX info,
+	const uint32_t info_id)
+{
+	static uint32_t l1i_counter = 0;
+	static uint32_t l1d_counter = 0;
+	static uint32_t l2_counter = 0;
+	static uint32_t l3_counter = 0;
+
+	/* Count cache types for allocation at first. */
+	if (caches == NULL) {
+		switch(info->Cache.Level) {
+			case 1:
+				switch (info->Cache.Type) {
+					case CacheInstruction:
+						numbers_of_caches[cpuinfo_cache_level_1i]++;
+					break;
+					case CacheData:
+						numbers_of_caches[cpuinfo_cache_level_1d]++;
+					break;

```

Comment:
Clang-cl complains about the missing other two enum values:
"error: enumeration values 'CacheUnified' and 'CacheTrace' not handled in switch [-Werror,-Wswitch]"
Could you add these values as empty cases as well?